### PR TITLE
Adding Search controller and view

### PR DESCRIPTION
**Pull request by ula0156**
1) Added search controller, view and model.
2) Added new search box on layout page.
3) Home page displays hyperLinks of products.

### CODE REVIEW

File: onlineShop.core/Data/ICartProvider.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using onlineShop;
+using System.Collections.Generic;
+
+namespace onlineShop.Data
+{
+    public interface ICartProvider
+    {
+        Cart GetCart(string userIdentifier);
```
Comment (simionliviumihai) [inline onlineShop.core/Data/ICartProvider.cs:8]: Used sessionIdOrLogin for the other method

File: onlineShop.core/Data/InMemory/InMemoryCartProvider.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            }
+
+            return cart;
+        }
+
+        public void CleanUpCart(List<string> listOfSessionAndLoginIds)
+        {
+            foreach (string sessionOrLoginId in listOfSessionAndLoginIds)
+            {
+                // didn't understand how to remove from this dictionary
```
Comment (simionliviumihai) [inline onlineShop.core/Data/InMemory/InMemoryCartProvider.cs]: This is not finished

File: onlineShopWeb/Controllers/BooksController.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+using onlineShop.Data.Database;
+using onlineShop.Products;
+using onlineShop.ProductPickers;
+using onlineShopWeb.DataAccess;
+
+namespace onlineShopWeb.Controllers
+{
+    public class BooksController : Controller
+    {
+        //private ProductsModel db = new ProductsModel();
```
Comment (simionliviumihai) [inline onlineShopWeb/Controllers/BooksController.cs]: This is commented. Remove if not needed

File: onlineShopWeb/Controllers/BooksController.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            //convert list of type Products into the list of type Book
+            return listOfBooks.ConvertAll(x => (Book)x);
+        }
+
+        public ActionResult Details(Guid id)
+        {
+            if (id == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+            }
```
Comment (simionliviumihai) [inline onlineShopWeb/Controllers/BooksController.cs:38]: Add empty  line after }

File: onlineShopWeb/Controllers/CartController.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+using System.Linq;
+using System.Web.Mvc;
+
+namespace onlineShopWeb.Controllers
+{
+    public class CartController : Controller
+    {
+        public RedirectToRouteResult AddToCart(Guid id)
+        {
+            // HttpContext contains an information about current context
```
Comment (simionliviumihai) [inline onlineShopWeb/Controllers/CartController.cs:17]: Check if id is null

File: onlineShopWeb/Controllers/SearchController.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.Mvc;
+
+namespace onlineShopWeb.Controllers
+{
+    public class SearchController : Controller
+    {
+        // GET: Search
```
Comment (simionliviumihai) [inline onlineShopWeb/Controllers/SearchController.cs]: Why GET?

File: onlineShopWeb/Controllers/SongsController.cs [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+namespace onlineShopWeb.Controllers
+{
+    public class SongsController : Controller
+    {
+        public ActionResult Index()
+        {
+            var songModel = GetfSongs();
+            return View(songModel);
+        }
+        private List<Song> GetfSongs()
```
Comment (simionliviumihai) [inline onlineShopWeb/Controllers/SongsController.cs]: Typo: GetSongs instead of GetfSongs

File: onlineShopWeb/Models/HomeViewModels.cs [hunk 8]
```diff
@@ -10,4 +10,14 @@ public class HomeViewModel
         public List<Song> SongsList { get; set; }
         
     }
+
+    public class HomeViewModelBook
```
Comment (simionliviumihai) [inline onlineShopWeb/Models/HomeViewModels.cs]: Why do you need this class if it only has a Book inside? Why not use Book directly

File: onlineShopWeb/Models/HomeViewModels.cs [hunk 9]
```diff
@@ -10,4 +10,14 @@ public class HomeViewModel
...
+    {
+        public Book book { get; set; }
+    }
+
+    public class HomeViewModelSong
```
Comment (simionliviumihai) [inline onlineShopWeb/Models/HomeViewModels.cs]: Same comment as the book here

File: onlineShopWeb/Utility/IdentifierLoginUsers.cs [hunk 10]
```diff
@@ -0,0 +1,23 @@
+﻿using onlineShop;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+
+namespace onlineShopWeb.Utility
+{
+    public static class IdentifierLoginUsers
```
Comment (simionliviumihai) [inline onlineShopWeb/Utility/IdentifierLoginUsers.cs]: This class name is strange.

**Review by simionliviumihai — CHANGES_REQUESTED** [review]

File: onlineShop.core/Data/ICartProvider.cs (hunk 1 above)
Comment (ula0156) [reply onlineShop.core/Data/ICartProvider.cs:8]: done

File: onlineShopWeb/Controllers/BooksController.cs (hunk 3 above)
Comment (ula0156) [reply onlineShopWeb/Controllers/BooksController.cs]: done

File: onlineShopWeb/Controllers/BooksController.cs (hunk 4 above)
Comment (ula0156) [reply onlineShopWeb/Controllers/BooksController.cs:38]: done

File: onlineShopWeb/Controllers/CartController.cs (hunk 5 above)
Comment (ula0156) [reply onlineShopWeb/Controllers/CartController.cs:17]: done

File: onlineShopWeb/Controllers/SearchController.cs (hunk 6 above)
Comment (ula0156) [reply onlineShopWeb/Controllers/SearchController.cs]: removed

File: onlineShopWeb/Controllers/SongsController.cs (hunk 7 above)
Comment (ula0156) [reply onlineShopWeb/Controllers/SongsController.cs]: done

File: onlineShopWeb/Utility/IdentifierLoginUsers.cs (hunk 10 above)
Comment (ula0156) [reply onlineShopWeb/Utility/IdentifierLoginUsers.cs]: changed it.

File: onlineShopWeb/Models/HomeViewModels.cs (hunk 8 above)
Comment (ula0156) [reply onlineShopWeb/Models/HomeViewModels.cs]: I don't need it. Removed it.

File: onlineShopWeb/Models/HomeViewModels.cs (hunk 9 above)
Comment (ula0156) [reply onlineShopWeb/Models/HomeViewModels.cs]: I don't need it. Removed it.

File: onlineShop.core/Data/InMemory/InMemoryCartProvider.cs (hunk 2 above)
Comment (ula0156) [reply onlineShop.core/Data/InMemory/InMemoryCartProvider.cs]: done

**Pull request merged**